src/bin/ion/commands/beta/generate/templates/rust/nested_type.templ

```diff
@@ -95,15 +101,19 @@
                     {% if field.value_type | is_built_in_type == false  %}
                         self.{{ field.name | snake }}.write_to(writer)?;
                     {% else %}
-                        writer.write_{% if field.isl_type_name == "symbol" %}symbol{% else %}{{ field.value_type | lower }}{% endif %}(self.value)?;
+                        writer.write_{% if field.isl_type_name == "symbol" %}symbol{% else %}{{ field.value_type | lower }}{% endif %}(self.value.to_owned())?;
                     {% endif %}
                 {% endfor %}
             {% elif abstract_data_type is object and abstract_data_type is containing("Structure") %}
                 writer.step_in(IonType::Struct)?;
                 {% for field in fields %}
                 writer.set_field_name("{{ field.name }}");
                     {% if field.value_type | is_built_in_type == false %}
-                        self.{{ field.name | snake }}.write_to(writer)?;
+                        {% if field.value_type is containing("Vec") %}
```

Comment:
If I define an ISL type such as `FeatureVector` or `FlowVector`, this is probably going to do surprising things.

Reply:
You are right. which is why with the new model we can just check the data model type.